src/server/cluster/cluster_family.cc

```diff
@@ -87,9 +87,9 @@ ClusterShard ClusterFamily::GetEmulatedShardInfo(ConnectionContext* cntx) const
                                .port = static_cast<uint16_t>(replica.listening_port)});
     }
   } else {
-    Replica::Info replication_info = server_family_->GetReplicaInfo();
+    DCHECK(etl.is_master);
```

Comment:
isn't the above if master flow?

Reply:
You're right. So right that the py tests caught it as well :)